### DISCUSSION TITLE
AlertManager recording rules should allow a range for the graph

### DISCUSSION THREAD

**Discussion by MaxDiOrio** [grafana/grafana#70001] [Alerting]
I often create Loki recording rules since we process thousands of log lines and metrics graphs are slow or impossible to load.

When creating a recording rule, I ran into one query where I was getting the following error, even though the query was working as expected in Explore for the same time period.

Failed to evaluate queries and expressions: failed to execute query A: maximum of series (5000) reached for a single query

Looking into it, I determined that when creating an AlertManager rule, the sample graph it generates uses a default, unchangeable 10 minute range, even when you specify the range directly in the query.

This becomes a problem when the query you're running will only work within a shorter time frame.  It would be good to be able to set the graph range as you can in Grafana Managed.  

When creating Grafana Managed, you can specify the graph range.

Managed Rule, range set to now-1m
![image](https://github.com/grafana/grafana/assets/25855524/78a9fe74-7bfe-439a-aebe-efef95fc52de)

AlertManager Rule, range fixed to 10 minutes.
![image](https://github.com/grafana/grafana/assets/25855524/eff1a26f-cfe5-4ea6-aa53-7d25d3e2c772)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82648. Feel free to continue the discussion around this there. Thank you!
